app/handler/post.go

```diff
@@ -0,0 +1,266 @@
+	post = new(model.Post)
+	if len(param) == 0 {
+		for _, p := range []string{"post_id", "slug"} {
+			post = getPostFromContext(ctx, p)
+			if post != nil {
+				return post
+			}
+		}
+	}
+	var err error
+	switch param[0] {
+	case "post_id":
+		id, convErr := strconv.Atoi(ctx.Param("post_id"))
+		if convErr != nil {
+			handleErr(ctx, 500, convErr)
+			return nil
+		}
+		err = post.GetPostById(int64(id))
+	case "slug":
+		slug := ctx.Param("slug")
+		err = post.GetPostBySlug(slug)
+	}
+	if err != nil {
+		handleErr(ctx, 404, err)
+		return nil
+	}
+	return post
+}
+
+// APIPostHandler retrieves the post with the given ID.
+func APIPostHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx, "post_id")
+	ctx.JSON(NewAPISuccessResponse(post))
+}
+
+// APIPostSlugHandler retrieves the post with the given slug.
+func APIPostSlugHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx, "slug")
+	ctx.JSON(NewAPISuccessResponse(post))
+}
+
+// APIPostsHandler gets an array of posts of length <= limit, starting at offset.
+// To paginate through posts, increment offset by limit until the length of the
+// post array in the response is less than limit.
+func APIPostsHandler(offset, limit int) golf.HandlerFunc {
+	// offset, limit args are default values
+	return func(ctx *golf.Context) {
+		var posts []*model.Post
+		var err error
+		err = ctx.Request.ParseForm()
+		if err != nil {
+			ctx.SendStatus(http.StatusInternalServerError)
+			ctx.JSON(APIResponseBodyJSON{Status: NewErrorStatusJSON(err.Error())})
+			return
+		}
+		if q, _ := ctx.Query("offset"); q != "" {
+			offset, err = strconv.Atoi(q)
+		}
+		if err != nil {
+			ctx.SendStatus(http.StatusBadRequest)
+			ctx.JSON(APIResponseBodyJSON{Status: NewErrorStatusJSON(err.Error())})
+			return
+		}
+		if q, _ := ctx.Query("limit"); q != "" {
+			limit, err = strconv.Atoi(q)
+		}
+		if err != nil {
+			ctx.SendStatus(http.StatusBadRequest)
+			ctx.JSON(APIResponseBodyJSON{Status: NewErrorStatusJSON(err.Error())})
+			return
+		}
+		published, _ := ctx.Query("published")
+		switch published {
+		case "true":
+			posts, err = model.GetPublishedPosts(offset, limit)
+		case "false":
+			posts, err = model.GetUnpublishedPosts(offset, limit)
+		default:
+			posts, err = model.GetAllPosts(offset, limit)
+		}
+		if err != nil {
+			ctx.SendStatus(http.StatusInternalServerError)
+			ctx.JSON(APIResponseBodyJSON{Status: NewErrorStatusJSON(err.Error())})
+			return
+		}
+		ctx.JSON(NewAPISuccessResponse(posts))
+	}
+}
+
+// APIPostCommentsHandler gets the comments on the given post.
+func APIPostCommentsHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
+	}
+	comments := post.Comments()
+	ctx.JSON(NewAPISuccessResponse(comments))
+}
+
+// APIPostAuthorHandler gets the author of the given post.
+func APIPostAuthorHandler(ctx *golf.Context) {
+	post := getPostFromContext(ctx)
+	if post == nil {
+		return
```

Comment:
Same thoughts as my comment above, maybe we could make small function that sends 404s with a brief message describing which resource couldn't be found (like "No comments found because post could not be found", "No author found because post could not be found", etc)